src/test/nodestore/DatabaseShard_test.cpp

```diff
@@ -1185,6 +1191,320 @@ class DatabaseShard_test : public TestBase
         }
     }
 
+    void
+    testImportWithHistoricalPaths(
+        std::string const& backendType,
+        std::uint64_t const seedValue)
+    {
+        using namespace test::jtx;
+
+
+        // Test importing with multiple historical
+        // paths
+        {
+            beast::temp_dir shardDir;
+            std::array<beast::temp_dir, 4> historicalDirs;
+            std::array<boost::filesystem::path, 4> historicalPaths;
+
+            std::transform(
+                historicalDirs.begin(),
+                historicalDirs.end(),
+                historicalPaths.begin(),
+                [](const beast::temp_dir& dir) { return dir.path(); });
+
+            beast::temp_dir nodeDir;
+            auto c = testConfig(
+                "importWithHistoricalPaths",
+                backendType,
+                shardDir.path(),
+                nodeDir.path());
+
+            auto& historyPaths = c->section(SECTION_HISTORICAL_SHARD_PATHS);
+            historyPaths.append({historicalPaths[0].string(),
+                historicalPaths[1].string(),
+                historicalPaths[2].string(),
+                historicalPaths[3].string()});
+
+            Env env{*this, std::move(c)};
+            DatabaseShard* db = env.app().getShardStore();
+            Database& ndb = env.app().getNodeStore();
+            BEAST_EXPECT(db);
+
+            auto const ledgerCount = 30;
+
+            TestData data(seedValue, 4, ledgerCount);
+            if (!BEAST_EXPECT(data.makeLedgers(env)))
+                return;
+
+            for (std::uint32_t i = 0; i < ledgerCount * ledgersPerShard; ++i)
+                BEAST_EXPECT(saveLedger(ndb, *data.ledgers_[i]));
+
+            BEAST_EXPECT(db->getCompleteShards() == bitmask2Rangeset(0));
+
+            db->import(ndb);
+            for (std::uint32_t i = 1; i <= ledgerCount; ++i)
+                waitShard(*db, i);
+
+            BEAST_EXPECT(
+                db->getCompleteShards() ==
+                bitmask2Rangeset(0b1111111111111111111111111111110));
+
+            auto const mainPathCount = std::distance(
+                boost::filesystem::directory_iterator(shardDir.path()),
+                boost::filesystem::directory_iterator());
+
+            // Only the two most recent shards
+            // should be stored at the main path
+            BEAST_EXPECT(mainPathCount == 2);
+
+            auto const historicalPathCount = std::accumulate(
```

Comment:
This file needs to `#include <numeric>`

Reply:
The Windows build is broken without it. I put this branch on top of the PR that fixes the Travis builds, and you can see all the Windows builds fail. https://travis-ci.com/github/ximinez/rippled/builds/177519102

Reply:
Oh, good catch. Done.